backend/layers/processing/publish_revisions.py

```diff
@@ -0,0 +1,128 @@
+
+    def check_datasets(self, collection_version: CollectionVersionWithDatasets) -> List[Dict]:
+        """Check that all datasets have been migrated and are in a success state"""
+        errors = []
+        current_dataset_versions = self.get_published_dataset_ids_from_collection_id(collection_version.collection_id)
+
+        for dataset in collection_version.datasets:
+            dataset_id = dataset.dataset_id.id
+            dataset_version_id = dataset.version_id.id
+            if not self.check_dataset_is_latest_schema_version(dataset):
+                errors.append(
+                    {
+                        "message": "Dataset is not the latest schema version.",
+                        "dataset_version_id": dataset_version_id,
+                        "dataset_id": dataset_id,
+                    }
+                )
+            elif dataset.status.processing_status != DatasetProcessingStatus.SUCCESS:
+                errors.append(
+                    {
+                        "message": dataset.status.validation_message,
+                        "dataset_status": dataset.status.to_dict(),
+                        "dataset_version_id": dataset_version_id,
+                        "dataset_id": dataset_id,
+                    }
+                )
+            else:
+                # track migrated.h5ad files store under the currently published dataset version id in s3 for
+                # clean up.
+                current_dataset_version = current_dataset_versions.get(dataset_id)
+                if current_dataset_version:
+                    key_name = self.get_key_prefix(f"{current_dataset_version}/migrated.h5ad")
+                    self.object_keys_to_delete.append(key_name)
+        return errors
+
+    def run(self):
+        for collection_version in self.business_logic.get_collections(CollectionQueryFilter(is_published=False)):
+            if collection_version.is_unpublished_version():
+                _collection_version = self.business_logic.get_collection_version(collection_version.version_id)
+                errors = self.check_datasets(_collection_version)
+                if errors:
+                    self.logger.error(
+                        "Unable to publish collection version.",
+                        extra={
+                            "collection_id": collection_version.collection_id.id,
+                            "collection_version_id": collection_version.version_id.id,
+                            "errors": errors,
+                        },
+                    )
+                else:
+                    self.logger.info(
+                        "Publishing collection version.",
+                        extra={"collection_version_id": collection_version.version_id.id},
+                    )
+                    self.business_logic.publish_collection_version(collection_version.version_id)
+        self.cleanup()
+
+    def cleanup(self) -> None:
```

Comment:
The auto_publish = False step in the main schema migration batch job only skips publish, it still runs the cleanup. So this (and the other code tracking which objects to delete) I don't think are necessary